android-kit-base/src/main/java/com/mparticle/kits/KitManagerImpl.java

```diff
@@ -1523,13 +1523,23 @@ private void confirmEmail(
         boolean hasHashedEmail = hashedEmail != null && !hashedEmail.isEmpty();
 
         if ((hasEmail || hasHashedEmail) && user != null) {
+            MParticle.IdentityType selectedIdentityType = null;
+            try {
+                String identityTypeStr = (kitConfiguration != null)
+                        ? kitConfiguration.getHashedEmailUserIdentityType()
+                        : null;
+                if (identityTypeStr != null && !identityTypeStr.equalsIgnoreCase("Unknown")) {
```

Comment:
I was planning to change this to an empty string, or "unselected" because that makes more sense.  Regardless of what it changes too, this method should be more generic and not require an actual string,  but rather grab an identity type based on a string of 'other, other2, other3" etc, and if it is 'unknown', or '', then return null

Reply:
I think you can remove the check for `~identityTypeStr.equalsIgnoreCase("Unknown")` on line 1531, and just do 

```
selectedIdentityType = MParticle.IdentityType.valueOf(identityTypeStr);
} catch ... {
warn the user that the selected identity type isn't working.  
}
```

Then `selectedIdentityType` remains null from line 1526